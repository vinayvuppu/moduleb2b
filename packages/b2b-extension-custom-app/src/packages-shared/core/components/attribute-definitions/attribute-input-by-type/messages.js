import { defineMessages } from 'react-intl';

export default defineMessages({
  editAttributeWarning: {
    id: 'attributeInputByType.nestedAttributes.editAttributeWarning',
    description: 'Warning message shown when nested attribute cannot be edited',
    defaultMessage:
      'Please create your product first in order to be able to edit this attribute',
  },
  aboveFifthLevelWarning: {
    id: 'attributeInputByType.nestedAttributes.aboveFifthLevelWarning',
    description:
      'Warning message shown when nested attribute are above the fifth level',
    defaultMessage:
      'The Merchant Center currently supports the edition of nested attributes up to five levels only.',
  },
});
