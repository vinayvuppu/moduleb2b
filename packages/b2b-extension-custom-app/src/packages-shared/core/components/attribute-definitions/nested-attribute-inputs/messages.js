import { defineMessages } from 'react-intl';

export default defineMessages({
  editAttributeWarning: {
    id: 'NestedAttributes.editAttributeWarning',
    description: 'Warning message shown when nested attribute cannot be edited',
    defaultMessage:
      'Please create your product first in order to be able to edit this attribute',
  },
  emptySetItem: {
    id: 'NestedAttributes.tooltip.emptySetItem',
    description: 'Empty set item',
    defaultMessage: 'New',
  },
  hideSetItems: {
    id: 'NestedAttributes.tooltip.hideSetItems',
    description: 'Hide set items bottom',
    defaultMessage: 'Hide set items ({countOfSetItem})',
  },
  showSetItems: {
    id: 'NestedAttributes.tooltip.showSetItems',
    description: 'Show set items bottom',
    defaultMessage: 'Show set items ({countOfSetItem})',
  },
  editTooltip: {
    id: 'NestedAttributes.tooltip.edit',
    description: 'NestedAttributes edit tooltip',
    defaultMessage: 'Edit Attributes',
  },
  editTooltipDisabled: {
    id: 'NestedAttributes.tooltip.editTooltipDisabled',
    description: 'NestedAttributes edit tooltip (disabled)',
    defaultMessage: 'Please update this level in order to edit this attribute',
  },
  deleteSetItemTooltip: {
    id: 'NestedAttributes.tooltip.deleteSetItem',
    description: 'NestedAttributes delete SetItem tooltip',
    defaultMessage: 'Delete set item',
  },
  addSetItemTooltip: {
    id: 'NestedAttributes.tooltip.addSetItem',
    description: 'NestedAttributes add SetItem tooltip',
    defaultMessage: 'Add set item',
  },
  addSetItemTooltipDisabled: {
    id: 'NestedAttributes.tooltip.addSetItemTooltipDisabled',
    description: 'NestedAttributes add SetItem tooltip (disabled)',
    defaultMessage: 'Multiple empty set items cannot be added',
  },
  emptyProductType: {
    id: 'NestedAttributes.emptyProductType',
    description: 'message that shows when product type is empty',
    defaultMessage:
      'The product type referenced does not contain any attribute.',
  },
  referencedProductType: {
    id: 'NestedAttributes.referencedProductType',
    description: 'Product type referenced hint',
    defaultMessage: 'Product type referenced: {productTypeName}',
  },
});
