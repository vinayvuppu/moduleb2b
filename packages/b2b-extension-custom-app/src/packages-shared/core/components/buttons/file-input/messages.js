import { defineMessages } from 'react-intl';

export default defineMessages({
  chooseFile: {
    id: 'ButtonFileInput.chooseFile',
    description: 'The label text of the field for upload local file',
    defaultMessage: 'Choose file',
  },
});
