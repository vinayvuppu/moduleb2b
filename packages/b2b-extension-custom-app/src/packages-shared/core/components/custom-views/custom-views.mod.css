.no-shrink-wrapper {
  display: flex;
  flex-shrink: 0;
}
