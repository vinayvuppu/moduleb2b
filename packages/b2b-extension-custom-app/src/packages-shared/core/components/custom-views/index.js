export { default } from './custom-views';
