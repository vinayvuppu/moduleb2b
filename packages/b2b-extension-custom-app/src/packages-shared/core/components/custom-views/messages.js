import { defineMessages } from 'react-intl';

export default defineMessages({
  selectCustomView: {
    id: 'CustomViews.selectCustomView',
    description: 'The text shown infront of the dropdown',
    defaultMessage: 'My views',
  },
  successfulCreationNotification: {
    id: 'CustomViews.successfulCreationNotification',
    description: 'Text for the notification when a view was created.',
    defaultMessage: 'Your view has been successfully created.',
  },
});
