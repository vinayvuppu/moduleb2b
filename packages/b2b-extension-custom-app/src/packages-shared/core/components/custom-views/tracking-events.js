export const trackingEvents = {
  createCustomView: {
    category: 'PimSearch-List',
    action: 'createCustomView',
    label: 'MyViews_save',
  },
};

export default trackingEvents;
