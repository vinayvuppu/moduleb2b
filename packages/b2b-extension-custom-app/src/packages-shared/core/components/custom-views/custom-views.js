import PropTypes from 'prop-types';
import React from 'react';
import { injectIntl, FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import { compose } from 'recompose';
import { Text, Spacings } from '@commercetools-frontend/ui-kit';
import { DOMAINS } from '@commercetools-frontend/constants';
import * as globalActions from '@commercetools-frontend/actions-global';
import injectTracking from '../tracking/inject-tracking';
import CustomViewsOnboarding from '../custom-views-onboarding';
import CustomViewsDropdown from '../custom-views-dropdown';
import { withModalState } from '../modal-state-container';

import {
  CustomViewCreateModal,
  CustomViewRenameModal,
  CustomViewDeleteModal,
} from '../custom-view-modals';
import messages from './messages';
import trackingEvents from './tracking-events';
import styles from './custom-views.mod.css';

export const CustomViews = props => {
  const Onboarding = props.components?.Onboarding || CustomViewsOnboarding;

  return (
    <React.Fragment>
      <Spacings.Inline alignItems="center" scale="s">
        <Onboarding isDisabled={props.disableInfoIcon} />
        <Spacings.Inline alignItems="center" scale="s">
          <div className={styles['no-shrink-wrapper']}>
            <Text.Body isBold={true}>
              <FormattedMessage {...messages.selectCustomView} />
            </Text.Body>
          </div>
          <CustomViewsDropdown
            projectKey={props.projectKey}
            value={props.view}
            options={props.views}
            isDirty={props.hasUnsavedChanges}
            onSelect={props.onSelect}
            onCreate={props.creationModal.handleOpen}
            onEdit={props.editingModal.handleOpen}
            onDelete={props.deletionModal.handleOpen}
            onSave={props.onSave}
            onReset={props.onReset}
          />
        </Spacings.Inline>
      </Spacings.Inline>
      <CustomViewCreateModal
        draft={props.view}
        isOpen={props.creationModal.isOpen}
        onCancel={props.creationModal.handleClose}
        customInfoMessage={props.customCreateModalInfoMessage}
        customTitle={props.customCreateModalTitle}
        onConfirm={formValues => {
          props.onCreate({ ...props.view, ...formValues }).then(() => {
            props.creationModal.handleClose();

            const { category, action, label } = trackingEvents.createCustomView;
            props.track(category, action, label);

            props.showNotification({
              kind: 'success',
              domain: DOMAINS.SIDE,
              text: props.intl.formatMessage(
                messages.successfulCreationNotification
              ),
            });
          });
        }}
      />
      <CustomViewRenameModal
        draft={props.view}
        isOpen={props.editingModal.isOpen}
        onCancel={props.editingModal.handleClose}
        onConfirm={formValues => {
          props.onSave({ ...props.view, ...formValues }).then(() => {
            props.editingModal.handleClose();
          });
        }}
        customInfoMessage={props.customRenameModalInfoMessage}
        customTitle={props.customRenameModalTitle}
      />
      <CustomViewDeleteModal
        view={props.view}
        isOpen={props.deletionModal.isOpen}
        onCancel={props.deletionModal.handleClose}
        onConfirm={() => {
          props.onDelete().then(() => {
            props.deletionModal.handleClose();
          });
        }}
        customInfoMessage={props.customDeleteModalInfoMessage}
        customConfirmButtonLabel={props.customDeleteModalConfirmButtonLabel}
      />
    </React.Fragment>
  );
};
CustomViews.displayName = 'CustomViews';

CustomViews.propTypes = {
  projectKey: PropTypes.string.isRequired,
  customCreateModalInfoMessage: PropTypes.string,
  customCreateModalTitle: PropTypes.string,
  customRenameModalInfoMessage: PropTypes.string,
  customRenameModalTitle: PropTypes.string,
  customDeleteModalInfoMessage: PropTypes.array,
  customDeleteModalConfirmButtonLabel: PropTypes.string,
  hasUnsavedChanges: PropTypes.bool.isRequired,
  disableInfoIcon: PropTypes.bool,
  view: PropTypes.shape({
    id: PropTypes.string,
  }),
  views: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.objectOf(PropTypes.string).isRequired,
    })
  ),
  components: PropTypes.shape({
    Onboarding: PropTypes.component,
  }),
  onSelect: PropTypes.func.isRequired,
  onSave: PropTypes.func.isRequired,
  onCreate: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
  onReset: PropTypes.func.isRequired,
  // withModalState
  creationModal: PropTypes.shape({
    isOpen: PropTypes.bool.isRequired,
    handleOpen: PropTypes.func.isRequired,
    handleClose: PropTypes.func.isRequired,
  }).isRequired,
  // withModalState
  editingModal: PropTypes.shape({
    isOpen: PropTypes.bool.isRequired,
    handleOpen: PropTypes.func.isRequired,
    handleClose: PropTypes.func.isRequired,
  }).isRequired,
  // withModalState
  deletionModal: PropTypes.shape({
    isOpen: PropTypes.bool.isRequired,
    handleOpen: PropTypes.func.isRequired,
    handleClose: PropTypes.func.isRequired,
  }).isRequired,
  // connect
  showNotification: PropTypes.func.isRequired,
  // injectIntl
  intl: PropTypes.shape({
    formatMessage: PropTypes.func.isRequired,
  }).isRequired,
  // injectTracking
  track: PropTypes.func.isRequired,
};

const mapDispatchToProps = {
  showNotification: globalActions.showNotification,
};

export default compose(
  connect(null, mapDispatchToProps),
  injectIntl,
  withModalState('creationModal'),
  withModalState('editingModal'),
  withModalState('deletionModal'),
  injectTracking
)(CustomViews);
