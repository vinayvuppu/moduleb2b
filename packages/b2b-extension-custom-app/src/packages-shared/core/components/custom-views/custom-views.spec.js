import React from 'react';
import { shallow } from 'enzyme';
import { intlMock } from '../../../test-utils';
import {
  CustomViewCreateModal,
  CustomViewRenameModal,
  CustomViewDeleteModal,
} from '../custom-view-modals';
import CustomViewsDropdown from '../custom-views-dropdown';
import { CustomViews } from './custom-views';
import messages from './messages';

const createView = custom => ({
  id: 'foo-view-id',
  name: {
    de: 'test-de-name',
  },

  ...custom,
});

const createTestProps = customProps => ({
  projectKey: 'test-project-key',
  hasUnsavedChanges: false,
  view: createView(),
  views: [createView()],
  onDelete: jest.fn(() => Promise.resolve()).mockName('onDelete'),
  onSave: jest.fn(() => Promise.resolve()).mockName('onSave'),
  onCreate: jest.fn(() => Promise.resolve()).mockName('onCreate'),
  onSelect: jest.fn().mockName('onSelect'),
  onReset: jest.fn().mockName('onReset'),
  track: jest.fn(),
  // injectIntl
  creationModal: {
    isOpen: false,
    handleOpen: jest.fn().mockName('creationModal.handleOpen'),
    handleClose: jest.fn().mockName('creationModal.handleClose'),
  },
  editingModal: {
    isOpen: false,
    handleOpen: jest.fn().mockName('editingModal.handleOpen'),
    handleClose: jest.fn().mockName('editingModal.handleClose'),
  },
  deletionModal: {
    isOpen: false,
    handleOpen: jest.fn().mockName('deletionModal.handleOpen'),
    handleClose: jest.fn().mockName('deletionModal.handleClose'),
  },
  // injectIntl
  intl: intlMock,
  // connect
  showNotification: jest.fn(),

  ...customProps,
});

describe('rendering', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<CustomViews {...props} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <CustomViewCreateModal>', () => {
    expect(wrapper).toRender(CustomViewCreateModal);
  });

  describe('with unsaved changes', () => {
    beforeEach(() => {
      props = createTestProps({
        hasUnsavedChanges: true,
      });
      wrapper = shallow(<CustomViews {...props} />);
    });

    it('should render <CustomViewRenameModal>', () => {
      expect(wrapper).toRender(CustomViewRenameModal);
    });
  });
});

describe('callbacks', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<CustomViews {...props} />);
  });

  describe('<CustomViewsDropdown>', () => {
    describe('when creating', () => {
      beforeEach(() => {
        wrapper.find(CustomViewsDropdown).prop('onCreate')();
      });

      it('should open the creation modal', () => {
        expect(props.creationModal.handleOpen).toHaveBeenCalled();
      });
    });
  });

  describe('<CustomViewCreateModal>', () => {
    describe.skip('when creating', () => {
      beforeEach(async () => {
        await wrapper.find(CustomViewCreateModal).prop('onConfirm')();
      });

      it('should invoke `onCreate`', () => {
        expect(props.onCreate).toHaveBeenCalled();
      });

      it('should close the creation modal', () => {
        expect(props.creationModal.handleClose).toHaveBeenCalled();
      });

      it('should show notification', () => {
        expect(props.showNotification).toHaveBeenCalledWith(
          expect.objectContaining({
            kind: 'success',
            text: messages.successfulCreationNotification.id,
          })
        );
      });
    });
  });

  describe('<CustomViewRenameModal>', () => {
    describe('when saving', () => {
      beforeEach(() => {
        props = createTestProps();
        wrapper = shallow(<CustomViews {...props} />);

        wrapper.find(CustomViewRenameModal).prop('onConfirm')();
      });

      it('should invoke `onSave`', () => {
        expect(props.onSave).toHaveBeenCalled();
      });

      describe.skip('when saving resolves', () => {
        it('should close the editing modal', () => {
          expect(props.editingModal.handleClose).toHaveBeenCalled();
        });
      });
    });
  });

  describe('<CustomViewDeleteModal>', () => {
    describe('when deleting', () => {
      beforeEach(() => {
        props = createTestProps();
        wrapper = shallow(<CustomViews {...props} />);

        wrapper.find(CustomViewDeleteModal).prop('onConfirm')();
      });

      it('should invoke `onDelete`', () => {
        expect(props.onDelete).toHaveBeenCalled();
      });

      describe.skip('when deletion resolves', () => {
        it('should close the deletion modal', () => {
          expect(props.deletionModal.handleClose).toHaveBeenCalled();
        });
      });
    });
  });
});
