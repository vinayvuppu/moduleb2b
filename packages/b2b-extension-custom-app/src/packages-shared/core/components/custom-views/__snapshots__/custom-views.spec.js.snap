// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should match snapshot 1`] = `
<Fragment>
  <Inline
    alignItems="center"
    justifyContent="flex-start"
    scale="s"
  >
    <injectIntl(withModalState(CustomViewsOnboarding)) />
    <Inline
      alignItems="center"
      justifyContent="flex-start"
      scale="s"
    >
      <div
        className="no-shrink-wrapper"
      >
        <TextBody
          isBold={true}
        >
          <FormattedMessage
            defaultMessage="My views"
            description="The text shown infront of the dropdown"
            id="CustomViews.selectCustomView"
            values={Object {}}
          />
        </TextBody>
      </div>
      <injectIntl(injectTracking(CustomViewsDropdown))
        isDirty={false}
        onCreate={[MockFunction creationModal.handleOpen]}
        onDelete={[MockFunction deletionModal.handleOpen]}
        onEdit={[MockFunction editingModal.handleOpen]}
        onReset={[MockFunction onReset]}
        onSave={[MockFunction onSave]}
        onSelect={[MockFunction onSelect]}
        options={
          Array [
            Object {
              "id": "foo-view-id",
              "name": Object {
                "de": "test-de-name",
              },
            },
          ]
        }
        projectKey="test-project-key"
        value={
          Object {
            "id": "foo-view-id",
            "name": Object {
              "de": "test-de-name",
            },
          }
        }
      />
    </Inline>
  </Inline>
  <injectIntl(CustomViewCreateModal)
    draft={
      Object {
        "id": "foo-view-id",
        "name": Object {
          "de": "test-de-name",
        },
      }
    }
    isOpen={false}
    onCancel={[MockFunction creationModal.handleClose]}
    onConfirm={[Function]}
  />
  <injectIntl(CustomViewRenameModal)
    draft={
      Object {
        "id": "foo-view-id",
        "name": Object {
          "de": "test-de-name",
        },
      }
    }
    isOpen={false}
    onCancel={[MockFunction editingModal.handleClose]}
    onConfirm={[Function]}
  />
  <injectIntl(CustomViewDeleteModal)
    isOpen={false}
    onCancel={[MockFunction deletionModal.handleClose]}
    onConfirm={[Function]}
    view={
      Object {
        "id": "foo-view-id",
        "name": Object {
          "de": "test-de-name",
        },
      }
    }
  />
</Fragment>
`;
